irohad/main/irohad.cpp

```diff
@@ -279,6 +281,8 @@ int main(int argc, char *argv[]) {
           config.database_config->working_dbname,
           config.database_config->maintenance_dbname,
           log);
+      rdb_opt = std::make_unique<iroha::ametsuchi::RocksDbOptions>(
```

Comment:
Let new iroha be compatible with old config files. 
Now if we have no /database/path key in config it fails with message `Missing database configuration!` which is hard to understand 

1. Make iroha compatible with old pg_opt configs
2. Let irohad be compatible with current normal database config without options database/path database/type, default type is postgres
3. Validate /database/type to be one of rocksdb or postgres or not-exist. No other options acceptable.

Reply:
We can't hide this option with default value. Because we have no default database. User have to select one.

Reply:
After this PR, iroha must be able to run with config from 1.2.x. We still support older configs where option "database" does not exist. Please do not break compatibility.